### PR TITLE
sql: remove date/intervalstyle_enabled from code

### LINKED ISSUE #81529: sql: deprecate datestyle_enabled / intervalstyle_enabled
As of v22.1, the reasons for `datestyle_enabled` and `intervalstyle_enabled` no longer hold.

We should always assume this is `true` in the codebase, and removes all checks of these variables (they probably need to hang around for backwards compat).

Jira issue: CRDB-15100

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/81529
 
No release note is needed, since v22.1 already included a release note
about how these are both hardcoded to true and cannot be changed.

The session variables and cluster settings remain in the code, but are
marked as retired.

Release note: None

### CODE REVIEW

**Comment by otan** [comment]
```

[02:37:44][run] pkg/sql/sem/cast/cast.go:184:32: undefined: SessionOptions
[02:37:44][run] pkg/sql/sem/cast/cast.go:194:39: undefined: SessionOptions
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:132:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:431:4: unknown field 'intervalStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:444:4: unknown field 'intervalStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:486:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:517:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:536:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:756:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast_map.go:971:4: unknown field 'dateStyleAffected' in struct literal of type Cast
[02:37:44][run] pkg/sql/sem/cast/cast.go:184:32: too many errors
```

:blob_eye_grin:

**Review by mgartner — APPROVED** [review]
💯

File: pkg/sql/logictest/testdata/logic_test/as_of [hunk 1]
```diff
@@ -18,7 +18,7 @@ SELECT * FROM t AS OF SYSTEM TIME INTERVAL '-1us'
 
 # Verify that we can use computed expressions.
 query I
-SELECT * FROM t AS OF SYSTEM TIME -( ('1000' || 'us')::INTERVAL )
+SELECT * FROM t AS OF SYSTEM TIME -( parse_interval('1000' || 'us') )
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/as_of:21]: did this have to change?

**Review by otan — APPROVED** [review]

File: pkg/sql/logictest/testdata/logic_test/as_of (hunk 1 above)
Comment (otan) [reply pkg/sql/logictest/testdata/logic_test/as_of:21]: oh, do as of statements have to be interval? TIL

**Comment by rafiss** [comment]
tftr!

bors r=otan,mgartner

**Pull request merged**